crates/mem-pool/src/pool.rs

```diff
@@ -341,8 +376,21 @@ impl MemPool {
     }
 
     /// output mem block
-    pub fn output_mem_block(&self) -> Result<(Option<CollectedCustodianCells>, BlockParam)> {
+    pub fn output_mem_block(
+        &self,
+        output_param: &OutputParam,
+    ) -> Result<(Option<CollectedCustodianCells>, BlockParam)> {
         let db = self.store.begin_transaction();
+
+        let (mem_block, post_merkle_state) = if output_param.retry_count > 0 {
+            self.repackage_mem_block(output_param)?
+        } else {
+            let mem_block = self.mem_block.clone();
+            let mem_db_state = self.fetch_state_db(&db)?;
+
+            (mem_block, mem_db_state.state_tree()?.merkle_state()?)
+        };
+
```

Comment:
Since we do not need to call `self.mem_block`, we can refactoring the code below into another function?